pkg/retry/retry.go

```diff
@@ -14,70 +14,70 @@
  * limitations under the License.
  */
 
-package cmd
+package retry
 
 import (
 	"context"
+	"math"
 	"math/rand"
-	"sync"
 	"time"
 )
 
-// lockedRandSource provides protected rand source, implements rand.Source interface.
-type lockedRandSource struct {
-	lk  sync.Mutex
-	src rand.Source
-}
-
-// Int63 returns a non-negative pseudo-random 63-bit integer as an
-// int64.
-func (r *lockedRandSource) Int63() (n int64) {
-	r.lk.Lock()
-	n = r.src.Int63()
-	r.lk.Unlock()
-	return
-}
-
-// Seed uses the provided seed value to initialize the generator to a
-// deterministic state.
-func (r *lockedRandSource) Seed(seed int64) {
-	r.lk.Lock()
-	r.src.Seed(seed)
-	r.lk.Unlock()
-}
-
 // MaxJitter will randomize over the full exponential backoff time
 const MaxJitter = 1.0
 
 // NoJitter disables the use of jitter for randomizing the
 // exponential backoff time
 const NoJitter = 0.0
 
-// Global random source for fetching random values.
-var globalRandomSource = rand.New(&lockedRandSource{
-	src: rand.NewSource(UTCNow().UnixNano()),
-})
+// Timer interface implements helper interface for defaultTimer
+type Timer interface {
+	Start(duration time.Duration)
```

Comment:
Timer.Reset() is tricky (https://golang.org/pkg/time/#Timer.Reset), we need to add this comment:

```
Cannot be re-called only on stopped or expired timers with drained channels. Cannot be used concurrently.
```

Reply:
It's internal @vadmeste we know what we are doing